### ISSUE TITLE
`Implementation-*` and `Specification-*` are missing in `MANIFEST.MF`

### ISSUE THREAD

**Issue by vy** [bndtools/bnd#5855]
[Log4j uses `bnd-maven-plugin` version `6.4.0` to generate OSGi and JPMS descriptors.](/apache/logging-parent/blob/fc9cc55f0c362a72d298c19a6fd89cc2baf4ebc9/pom.xml#L716) Though [users noticed that generated `MANIFEST.MF` files are missing `Implementation-*` and `Specification-*` entries](/apache/logging-log4j2/issues/1923), which were earlier [populated by `maven-archiver` by default](https://maven.apache.org/shared/maven-archiver/#class_manifest). While we can fix the problem on our side manually, I have the impression that this should ideally be implemented by `bnd-maven-plugin` and made available to everyone. What do you think?

List of entries we are particularly interested in:
* `Implementation-Title`
* `Implementation-Vendor`
* `Implementation-Version`
* `Specification-Title`
* `Specification-Vendor`
* `Specification-Version`

**Comment by pkriens** [comment]
You can add these headers in the bnd file. Since the properties are processed by the macro pre-processor, you can deduce some of the information from the Bundle-Vendor or Bundle-Version.
```bnd
    ....
    Implementation-Title: Apache ...
    Implementation-Vendor Apache ...
    Implementation-Version ${Bundle-Version}
    Specification-Title Apache ...
    Specification-Vendor Apache ...
    Specification-Version 1.2.3
```

I am not sure how the maven-archiver can 'calculate' those values except maybe the Implementation-Version from the  artifact version?

**Comment by rotty3000** [comment]
https://github.com/apache/maven-archiver/blob/cc2f6a219f6563f450b0c00e8ccd651520b67406/src/main/java/org/apache/maven/archiver/MavenArchiver.java#L443

**Comment by pkriens** [comment]
I think this should be handled in the bnd file. The macros allow you to not have redundancy. 

I do not think we should automate this in any way, I am pretty uncomfortable with the choices the maven archiver made. Fine it people want it, but setting the headers in the bnd file allows them to do the same.

**Comment by pkriens** [comment]
I'd like to close this issue unless there is a very good reason to not close it. Ok?

**Comment by vy** [comment]
@pkriens, we can indeed fix this in our BND file and [we already acknowledged that relying on these can be considered as a bad practice](https://github.com/apache/logging-log4j2/issues/1923#issuecomment-1786818254). Nevertheless, this is what every single Maven-packaged project out there in the wild contains and there are users who still insist on using it. Hence, either you or the downstream (Log4j, etc.) will end up implementing this hack for backward compatibility reasons. I thought fixing this at the source would make the life of the downstream easier.

In conclusion, yes, I am asking you to implement a hack in `bnd-maven-plugin` for backward compatibility concerns of the downstream. Yes, the downstream can live without this hack. I leave the decision to you. I am fine with either outcomes.

**Comment by rotty3000** [comment]
@vy maybe send a PR? And, like in archiver, if you make it opt-in, there wouldn't likely be strong argument against merging it.

**Comment by chrisrueger** [comment]
> I have implemented [the BND fix in `logging-parent`](apache/logging-parent@a43a0b26066b637a5b0cd07cee877c83f0f3347d),

As the link was broken: I assume it should be this: https://github.com/apache/logging-parent/commit/a43a0b26066b637a5b0cd07cee877c83f0f3347d

**Comment by vy** [comment]
@chrisrueger, right. Thanks. Fixed the comment.
